### PR TITLE
Fix inconsistencies introduced in side-effects of StoreTailer#approximateLastCycle2(int)

### LINKED ISSUE #1532: toEnd doesn't work when the last roll cycle is empty
If the last roll cycle in the queue is empty, `toEnd` is broken.

This is a result of the internal state of the Tailer being made inconsistent by the `approximateLastCycle2` scan.

### PR DESCRIPTION
**Pull request by nicktindall**
Fixes #1532

### CODE REVIEW

File: src/main/java/net/openhft/chronicle/queue/impl/single/StoreTailer.java [hunk 1]
```diff
@@ -783,6 +783,7 @@ private long approximateLastCycle2(int lastCycle) throws StreamCorruptedExceptio
             if (this.store != wireStore) {
                 releaseStore();
                 this.store = wireStore;
+                this.cycle = lastCycle;
```
Comment (nicktindall) [inline src/main/java/net/openhft/chronicle/queue/impl/single/StoreTailer.java:786]: We used to update the `store` field without making a corresponding change to the `cycle` field.

**Review by JerryShea — APPROVED** [review]

**Comment by JerryShea** [comment]
love it!

File: src/main/java/net/openhft/chronicle/queue/impl/single/StoreTailer.java (hunk 1 above)
Comment (nicktindall) [reply src/main/java/net/openhft/chronicle/queue/impl/single/StoreTailer.java:786]: this caused issues for the logic above that says:

` wireStore = (cycle == lastCycle) ? this.store : queue.storeForCycle(lastCycle, queue.epoch(), false, this.store);`

on the subsequent recursion.

**Comment by tgd** [comment]
Great work Nick

**Pull request merged**